tests/jballerina-unit-test/src/test/resources/test-src/types/byte/byte_as_int_test.bal

```diff
@@ -186,6 +189,22 @@ function testDowncastOfByteArrayCastToIntArray() {
     assertEquality(255, barr2[3]);
 }
 
+function testByteArrayLiteralCastToReadOnlyType() {
+    byte[] & readonly arr1 = <byte[] & readonly>base16 `aa bb`;
+    assertTrue(arr1.isReadOnly());
+
+    int[2] & readonly arr2 = <int[2] & readonly>base16 `aa bb`;
+    assertTrue(arr2.isReadOnly());
+
+    byte[2] & readonly arr3 = <readonly>base16 `aa bb`;
+    assertTrue(arr3.isReadOnly());
+
+    anydata[] & readonly arr4 = <anydata[] & readonly>base16 `aa bb`;
+    assertTrue(arr4.isReadOnly());
+
+    any[] arr5 = <any[] & readonly>base16 `aa bb`;
```

Comment:
Not asserted/used?

Reply:
Updated a41828e66acf222373ef93deff481bde6310fd77